cns/restserver/util.go

```diff
@@ -235,29 +236,52 @@ func (service *HTTPRestService) updateIpConfigsStateUntransacted(req cns.CreateN
 		}
 	}
 
-	// Add the newIpConfigs, ignore if ip state is already in the map
-	service.addIPConfigStateUntransacted(req.NetworkContainerid, newIPConfigs)
+	// Add the newIpConfigs for 3 different conditions
+	// 1. If NC version from NMAgent is larger or same as new NC version, add new IP as available.
+	// 2. If NC version from NMAgent is larger or same as existing NC version, set existing pending programming IP as available and add new IP as pending programming.
+	// 3. If NC version from NMAgent is smaller than existing NC version, add new IP as pending programming.
+	var newNCVersion int
+	var existingNCVersion int
+	newNCVersion, _ = strconv.Atoi(req.Version)
+	existingNCVersion, _ = strconv.Atoi(existingNCVersionInString)
+	// Wait for Ashvin's thread update which querying NMAgent to get latest NC version and save it locally.
+	nmAgentNCVersion := getNCVersionFromNMAgent(req.NetworkContainerid)
```

Comment:
Again, I dont think we should be able to call NMAgent while acquiring the Service lock. This function should compare the NCVersion with existing NMAgent version (persisted in ContainerStatus) and if it is higher than set the ips to PendingProgramming and exit.

Reply:
Instead of calling getNCVersionFromNMAgent, you should pull HostVersion from ContainerStatus.HostVersion. That is only required in case updated NCVersion is same as HostVersion, in that case we will set the ips to Available state, else Ips are in PendingState if not already set

Reply:
getNCVersionFromNMAgent didn't have any implementation yet. It could be the function which fetch NMAgent version locally from what background thread got and saved.
Whether it's get from ContainerStatus.HostVersion or not, it depends on Ashvin's PR  that where will he save the NMAgent version.